Arranger Dev Documentation Update (#892)
overture-stack/arranger: docs/setup.md

# Setup

## Prerequisites

Before you begin, ensure you have the following installed on your system:

- Node.js (v18)
- npm (v8.3.0 or higher)
- [Docker](https://www.docker.com/products/docker-desktop/) (v4.32.0 or higher)

## Developer Setup

This guide will walk you through setting up a complete development environment, including Arranger and its complementary services.

### Setting up supporting services

> We'll use our Conductor service, a flexible Docker Compose setup, to spin up Maestro's complementary services.

Reviewer: "we" here implies us + the users reading the guide, therefor "our" would mean Conductor is also theirs. Mine!
```suggestion
We'll use the Overture Conductor service, a flexible Docker Compose setup, to spin up Maestro's complementary services.
```

1. Clone the Conductor repository and navigate to its directory:

    ```bash
    git clone https://github.com/overture-stack/conductor.git
    cd conductor
    ```

2. Run the appropriate start command for your operating system:

    | Operating System | Command |
    |------------------|---------|
    | Unix/macOS       | `make arrangerDev` |
    | Windows          | `make.bat arrangerDev` |

    <details>
    <summary>**Click here for a detailed breakdown**</summary>

    This command will set up all complementary services for Arranger development as follows:

    ![arrangerDev](./assets/arrangerDev.svg 'Arranger Dev Environment')

    | Service | Port | Description | Purpose in Arranger Development |
    |---------|------|-------------|------------------------------|
    | Conductor | `9204` | Orchestrates deployments and environment setups | Manages the overall development environment |
    | Elasticsearch | `9200` | Distributed search and analytics engine | Provides fast and scalable search capabilities over indexed data |
    | Stage | `3000` | Web Portal Scaffolding | Houses Arranger's search UI components |

    :::note Arranger uses Elasticsearch 7 

>     Our search platform is built on and compatible with version 7.x of Elasticsearch. Applications and queries need to follow Elasticsearch 7 syntax and conventions.

Reviewer: This notice was already in the overview, and while it's important to note they need to set up v7 here, using queries is not in the scope of "setup", strictly speaking.

I've amended this to match the changes I've suggested in the overview, but I'd advocate for removing this paragraph here.
```suggestion
    Our search platform is built on and compatible with version 7.x of Elasticsearch. All queries to ES must follow that version's syntax and conventions.
```

        ```
        # ==============================
        # Arranger Environment Variables
        # ==============================

        # Arranger Variables
        ENABLE_LOGS=false
        # Elasticsearch Variables
        ES_HOST=http://elasticsearch:9200
        ES_USER=elastic
        ES_PASS=myelasticpassword
        # Stage Variables
        REACT_APP_BASE_URL=http://localhost:3000
        ```

        <details>
          <summary>**Click here for a detailed explanation of Arranger's environment variables**</summary>

          **Arranger Variables**
          - `ENABLE_LOGS`: Determines whether logging is enabled
          - **Elasticsearch Variables**
          - `ES_HOST`: The URL of your Elasticsearch instance
          - `ES_USER` and `ES_PASS`: The credentials for accessing Elasticsearch

          **Stage Variables**

          - `REACT_APP_BASE_URL`: The base URL for your front-end application (Stage)
          - `REACT_APP_ARRANGER_ADMIN_ROOT`: The URL for the Arranger GraphQL endpoint
        </details>

    :::

3. Install the required npm packages:

    ```bash
    npm ci
    ```

    :::warning

>     Python version 3.10.15 or lower is required. Python 3.12 is not supported.

Reviewer: ```suggestion
    Python version 3.10.15 or lower is required. Python 3.12 is not supported yet.
```

    :::

4. Bootstrap the Arranger repository:

    ```bash
    npm run bootstrap
    ```

5. Run the Arranger server:

    ```bash
    npm run server
    ```

Once the server starts, you can access Arranger-Server at `http://localhost:5050/graphql`.

### Running the Arranger Components

> :::info Coming Soong

Reviewer: ```suggestion
:::info Coming Soon
```

> We are currently working on updating our development environment for Arrangers components. Documentation for running Arranger components, including the developer setup and Storybook integration, will be available here in a our next update.

Reviewer: ```suggestion
We are currently working on updating our development environment for Arranger Components. Documentation for implementing them, including their development setup and Storybook integration, will be available here in the near future.
```
